sql/catalyst/src/main/scala/org/apache/spark/sql/types/GenericArrayData.scala

```diff
@@ -0,0 +1,127 @@
+package org.apache.spark.sql.types
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.SpecializedGetters
+import org.apache.spark.unsafe.types.{UTF8String, Interval}
+
+abstract class ArrayData extends SpecializedGetters {
+  def array(): Array[Any]
+
+  def get(ordinal: Int): Any
```

Comment:
the generic `get` and `array` method is useful in interpreted mode, but it's a lot painful to handle `Any` between scala and java, so I put this interface in scala.